### ISSUE TITLE
Console Command Request: CSV Wallet Transactions Export

### ISSUE THREAD

**Issue by Rock-N-Troll** [Veil-Project/veil#518]
This function would output a csv file in a separate directory in the wallet much like the backups have their own folder.

exporttransactions [type] range

Where type is a json array that can be used to specify which transaction types to export (as defined in the wallet).

Where range would be a date range OR the most recent X transactions. TBD.

The default would be all transaction types and all range.

Users want to see their data and use their data to analyze what is going on in their own wallets. Additionally, this can help people track their Veil much more closely (without giving up privacy). 

This command could also be useful if a user chooses to use a third party application that helps them keep track of their portfolio, their own transaction history, or other creative use of this data for their own benefit.

**Comment by Rock-N-Troll** [comment]
CSV makes the most sense for file output whereas JSON might be better for output in the console. I think most users will want CSV output and general people using the API will want JSON. This might be best to be 2 tasks stemming from 1 task if that approach is useful.

**Comment by veilgets** [comment]
Bounty: 583.33

**Comment by veilgets** [comment]
Developer: MatWaller

**Comment by seanPhill** [comment]
FYI. I use a Python script to convert JSON to CSV and I have to separate certain transactions (orphans/zero confirmations, sent and ones with a basecoin address, from the most common 'received' txs), because the columns get messed up when some of the transactions have some fields that don't exist on the others. Maybe you can solve that, but I just use three or four sheets instead, the fourth being orphans, which are only of transient interest.

**Comment by CaveSpectre11** [comment]
Developer: darkeseed

**Comment by eugene-so** [comment]
Please clarify what we are actually trying to export to CSV.  In the file walletbalances.h, i can see their are veil, CT, RingtCT and Zerocoin balances.  Is this what we are interested in dumping?  Thanks.

**Comment by seanPhill** [comment]
> Please clarify what we are actually trying to export to CSV. In the file walletbalances.h, i can see their are veil, CT, RingtCT and Zerocoin balances. Is this what we are interested in dumping? Thanks.

 I just ran exporttransactions on one of my testnet wallets (that was fresh on May 2nd) and it is definitely listing transactions that are not only mined coins, but staked, sent, received amounts, and orphans too.

It generated over 55,000 lines in a 15.4 MB CSV file, with human readable formatted dates, in a quite reasonably fast time. Hardly noticeable. About 13 seconds, for really a lot of transactions.

This is perfect! It's not only basecoins.
Maybe this issue can now be closed. I don't know what to do about the merge 904 being described as basecoin transactions only.

@Rock-N-Troll are you happy with using `listtransactions` for JSON output (without human readable dates) in the Console, while using `exporttransactions` for CSV file output and nice, human readable dates, as we have? And as this has been merged for nearly a year, and included in a release wallet since January, are you happy for this issue to be closed?
